### PR TITLE
Rt - role to install RT and a number of extensions

### PR DESCRIPTION
**Pull request by mattrm**
Nigel - can you test this ansible play on a new RT server for me?  It's potentially going to need some more work, but I'd like you to test this out first please.

### CODE REVIEW

File: ansible/roles/rt/tasks/main.yml [hunk 1]
```diff
@@ -1,38 +1,147 @@
...
-- name: create docker rt server
-  command: docker run -d -e="RT_DATABASE_PW={{rt_database_pw}}" -v="/data/rt:/data" -p=127.0.0.1:5025:25 -p=127.0.0.1:5080:80 -link=rt_pg:db -name={{ docker_container }} okfn/rt:0.16
-  when: rt.rc == 1
-
-- name: configure docker rt server
-  copy: src=RT_SiteConfig.pm dest=/data/rt/rt_conf/ owner=root group=www-data mode=0640
-  notify: restart rt
+
+- name: Install RT pre-reqs through apt-get
+  apt: pkg={{ item }} state=installed
```
Comment (nigelbabu) [inline ansible/roles/rt/tasks/main.yml]: Add `update_cache=yes`

File: ansible/roles/rt/tasks/main.yml [hunk 2]
```diff
@@ -1,38 +1,147 @@
...
+  - libyaml-appconfig-perl 
+  - libssl-dev 
+  - libexpat1-dev
+  - libdbd-mysql-perl
+  - libnet-smtp-ssl-perl 
+  - libdatetime-event-ical-perl
+  - procmail
+
+- name: get latest RT release
+  get_url: url=https://download.bestpractical.com/pub/rt/release/rt-4.2.9.tar.gz dest=/opt/
```
Comment (nigelbabu) [inline ansible/roles/rt/tasks/main.yml]: Can you make the version number a variable? Makes it easier to upgrade in the future.

File: ansible/roles/rt/tasks/main.yml [hunk 3]
```diff
@@ -1,38 +1,147 @@
...
+- name: configure RT install
+  command: ./configure
+  args:  
+    chdir: /opt/rt4-install
+
+- name: configure cpan
+  command: perl -MCPAN -e 'my $c = "CPAN::HandleConfig"; $c->load(doit => 1, autoconfig => 1); $c->edit(prerequisites_policy => "follow"); $c->edit(build_requires_install_policy => "yes"); $c->commit'
+  args: 
+    chdir: /opt/rt4-install
+  async: 3600
```
Comment (nigelbabu) [inline ansible/roles/rt/tasks/main.yml]: Does it really need 60 minutes?

File: ansible/roles/rt/tasks/main.yml [hunk 4]
```diff
@@ -1,38 +1,147 @@
...
+    - rt-extension-nagios
+    - rt-extension-mergeusers
+    - rt-extension-repeatticket
+    - rt-extension-resetpassword
+    - rt-extension-spawnlinkedticketinqueue
+
+- name: get nagios all nagios plugins
+  get_url: url={{ item.url }} dest=/opt/{{ item.name }}/master.zip
+  with_items:
+    - { url: 'https://github.com/bestpractical/rt-extension-activityreports/archive/master.zip', name: 'RT-Extension-ActivityReports' }
```
Comment (nigelbabu) [inline ansible/roles/rt/tasks/main.yml]: Pulling master for everything is problematic. There may be something in the future that'll break with our version of RT. Is there a way to fix a commit? Or perhaps git clone it to a particular commit?

File: ansible/roles/rt/tasks/main.yml (hunk 3 above)
Comment (mattrm) [inline ansible/roles/rt/tasks/main.yml]: Probably not, but it was failing on smaller numbers, so I just put a large number in there to make it work.  It is slow however.  Just cause it says 60 mins in there, doesn't make the role take 60 mins.

File: ansible/roles/rt/tasks/main.yml (hunk 4 above)
Comment (mattrm) [inline ansible/roles/rt/tasks/main.yml]: I agree, it's a poor way of doing it which probably break in the future.  The extensions are pretty basic and don't tend to get released much generally.  Would git cloning all of them be better then?

File: ansible/roles/rt/tasks/main.yml [hunk 5]
```diff
@@ -1,38 +1,171 @@
...
+  - libdbd-mysql-perl
+  - libnet-smtp-ssl-perl 
+  - libdatetime-event-ical-perl
+  - libdbd-pg-perl
+  - procmail
+  - spawn-fcgi
+
+- name: check if RT is installed
+  stat: path=/opt/rt4
+  register: rt_exists
```
Comment (nigelbabu) [inline ansible/roles/rt/tasks/main.yml]: Worth switching to the .installed file method instead.

File: ansible/roles/rt/templates/RT_SiteConfig.pm [hunk 6]
```diff
@@ -4,8 +4,13 @@ Set( $WebPath, "" );
 Set( $WebPort, 443 );
 Set( $WebDomain, "rt.okfn.org" );
 Set( $WebSecureCookies, 1 );
-Set( $DatabaseHost, $ENV{DB_PORT_5432_TCP_ADDR} );
-Set( $DatabasePassword, $ENV{RT_DATABASE_PW} );
+
+Set( $DatabaseType, "Pg" );
+Set( $DatabaseHost, "db001.cxglaymtrf0h.eu-west-1.rds.amazonaws.com" );
```
Comment (nigelbabu) [inline ansible/roles/rt/templates/RT_SiteConfig.pm]: Please use a variable.

File: ansible/roles/rt/tasks/main.yml [hunk 7]
```diff
@@ -1,38 +1,175 @@
...
+  unarchive: src=/opt/rt-{{ rt_version }}.tar.gz dest=/opt/ copy=no
+  when: rt_exists.stat.exists == false
+
+- name: copy rt dir to rt4 dir
+  command: mv /opt/rt-{{ rt_version }} /opt/rt4-install
+  register: rtdirectory
+  when: rt_exists.stat.exists == false
+
+- name: configure RT install
+  {command: ./configure
```
Comment (nigelbabu) [inline ansible/roles/rt/tasks/main.yml]: Is that `{` supposed to be there or a typo?

File: ansible/roles/rt/tasks/main.yml [hunk 8]
```diff
@@ -1,38 +1,175 @@
...
+  - libyaml-perl 
+  - libyaml-appconfig-perl 
+  - libssl-dev 
+  - libexpat1-dev
+  - libdbd-mysql-perl
+  - libnet-smtp-ssl-perl 
+  - libdatetime-event-ical-perl
+  - libdbd-pg-perl
+  - procmail
+  - spawn-fcgi
```
Comment (nigelbabu) [inline ansible/roles/rt/tasks/main.yml]: You need to add `unzip` to the list.

File: ansible/roles/rt/tasks/main.yml [hunk 9]
```diff
@@ -1,38 +1,175 @@
...
+
+# Configure Procmail
 
 - name: configure procmail
-  copy: src=procmailrc.rt dest=/data/rt/rt_postfix/procmailrc.rt owner=root group=root mode=0644
+  copy: src=procmailrc.rt dest=/opt/rt4/etc/procmailrc.rt owner=root group=root mode=0644
+  when: rt_exists.stat.exists == false
+
+- name: Create .installed file now that RT is installed
+  file: path=/opt/rt4/.installed state=file
```
Comment (nigelbabu) [inline ansible/roles/rt/tasks/main.yml]: This needs to be `state=touch`. `state=file` will NOT create a new file if it doesn't exist.

**Comment by nigelbabu** [comment]
Merged this in manually.

**Pull request closed without merging**